### PR TITLE
Add Matrix tests and fix compat bugs

### PR DESCRIPTION
**Pull request by hughbe**
Related: https://github.com/dotnet/corefx/pull/24238

### CODE REVIEW

File: tests/testmatrix.c [hunk 1]
```diff
@@ -0,0 +1,799 @@
...
+	assert (points[1].Y == 28);
+	assert (points[2].X == 28);
+	assert (points[2].Y == 40);
+
+	// Negative tests.
+	status = GdipTransformMatrixPointsI (NULL, points, 3);
+	assert (status == InvalidParameter);
+
+	// This causes a null pointer dereference in GDI+.
+#if !defined(USE_WINDOWS_LIBGDIPLUS)
```
Comment (hughbe) [inline tests/testmatrix.c]: Another GDI+ bug that I don't think is worth replicating

File: tests/testmatrix.c [hunk 2]
```diff
@@ -679,6 +679,7 @@ static void test_isMatrixIdentity ()
 	assert (isIdentity == TRUE);
 
 	// Less close to identity.
+	// FIXME: it appears that GDI+ is less precise than GDI+.
```
Comment (hughbe) [inline tests/testmatrix.c]: @mellinoe @akoeplinger do you know the algorithm GDI+ uses for `GdipIsMatrixIdentity`? I've tried narrowing the values down to around `+-0.99949f` but I don't want to start mucking around with things that potentially break

Comment (mellinoe) [reply tests/testmatrix.c]: Sorry, I don't know -- I'm really not familiar with the GDI+ codebase.

Comment (hughbe) [reply tests/testmatrix.c]: No problem, just checking!

File: tests/testmatrix.c [hunk 3]
```diff
@@ -0,0 +1,799 @@
...
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+int floatsEqual(float v1, float v2)
+{
+		return fabs (v1 - v2) < 0.0001;
+}
+
+void verifyMatrix (GpMatrix *matrix, REAL e1, REAL e2, REAL e3, REAL e4, REAL e5, REAL e6)
```
Comment (akoeplinger) [inline tests/testmatrix.c]: this and floatsEqual already exist in testhelpers.h, please just #include that and use it from there (and while you're at it please get rid of the copy in testpathgradientbrush.c as well).

File: tests/Makefile.am [hunk 4]
```diff
@@ -110,12 +116,13 @@ testpng_SOURCES =		\
 EXTRA_DIST =			\
 	$(testgdi_SOURCES)	\
 	$(testbits_SOURCES)	\
-	$(testbrush_SOURCES)	\
+	$(testbrush_SOURCES) \
```
Comment (akoeplinger) [inline tests/Makefile.am]: nit: revert the whitespace changes here and below.

**Comment by hughbe** [comment]
Good point - I usually copy-paste the code into rextester.com to test the GDI+ code against Windows (then also on my local machine at the end to make sure). Rextester.com doesn't allow including files so I just copy-pasted them and forgot to revert that.

I've addressed the PR feedback, thanks for reviewing!

**Pull request merged**